packages/console/src/notebook/ScriptEditor.tsx

```diff
@@ -22,24 +54,24 @@ class ScriptEditor extends Component {
     this.updateShortcuts = this.updateShortcuts.bind(this);
 
     this.contextActionCleanups = [];
-    this.completionCleanup = null;
-    this.editor = null;
     this.editorComponent = React.createRef();
 
     this.state = {
       model: null,
     };
   }
 
-  componentDidMount() {
+  componentDidMount(): void {
     ShortcutRegistry.addEventListener('onUpdate', this.updateShortcuts);
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps: ScriptEditorProps): void {
     const { sessionLanguage, settings } = this.props;
-    const { language } = settings;
-
-    const languageChanged = language !== prevProps.settings.language;
+    let language = '';
+    if (settings) {
+      language = settings.language;
+    }
```

Comment:
Why not just set it right away?
```suggestion
    const language = settings?.language;
```